mp/src/game/server/sdk/sdk_player.cpp

```diff
@@ -1523,6 +1523,47 @@ int CSDKPlayer::OnTakeDamage( const CTakeDamageInfo &inputInfo )
 			}
 		}
 
+		if ( info.GetDamageType() & DMG_BULLET ){ // the player has been shot
+
+			// apply damage falloff
+			// the weapon that did the hurting
+			CWeaponSDKBase* pWeapon = dynamic_cast<CWeaponSDKBase*>(info.GetWeapon());
+			CSDKWeaponInfo* pWeaponInfo = pWeapon?CSDKWeaponInfo::GetWeaponInfo(pWeapon->GetWeaponID()):NULL;
+			if( pWeaponInfo ){ // just in case
+				// get the distance from the attacker to self
+				Vector vecToAttacker = GetAbsOrigin() - info.GetAttacker()->GetAbsOrigin();
```

Comment:
It's a good approximation, but how about using the actual path the bullet travelled?